Sources/SourceControl/RepositoryManager.swift

```diff
@@ -243,7 +209,14 @@ public class RepositoryManager {
                             self.delegate?.handleWillUpdate(handle: handle)
                         }
 
-                        try repo.fetch()
+                        // FIXME: Workaround for `FileLock` only working on `LocaFileSystem`
```

Comment:
I think we should look into `FileLock` essentially being a no op when using the in-memory fs instead of doing this workaround.

Reply:
I can see two possible ways of achieving this:
1) make `FileLock` part of `FileSystem` --> More extensible
2) make `FileLock` aware of the `FileSystem` being used --> Fewer changes to the API required

Shouldn't `lock()` behave like a normal lock when using the in-memory fs?

Reply:
#1 here seems like the best solution, although it would of course involve more changes, as you point out.  It makes sense to me that for an in-memory file system, locking would be just normal locking (not a no-op).  For this case, it seems that maybe all that would be needed is a new function on `FileSystem` that lets you execute a block while holding a lock on a file, e.g. `withLock(on: path) { /* do something */ }`.  The local-file-system implementation is straightforward, but the in-memory file system implementation would involve a locked map of paths to FileLocks (though that's not technically sufficient either because of symlinks, but the implementation there could be refined over time).